### PR TITLE
Csharp renaming

### PR DESCRIPTION
**Pull request by nikoo28**
Persistence:
DynamoDbMetastorePersistenceImpl --> DynamoDbMetastoreImpl
JdbcMetastorePersistenceImpl --> JdbcMetastoreImpl
MemoryPersistenceImpl to InMemoryMetastoreImpl
IMetastorePersistence --> IMetastore
LoadLatestValue (method) --> LoadLatest

AppEncryption --> Session
AppEncryptionBytesImpl --> SessionBytesImpl
AppEncryptionJsonImpl --> SessionJsonImpl
AppEncryptionPartition --> Partition
AppEncryptionSessionFactory --> SessionFactory

Other changes:
Appencryption README
Reference app README
Reference app refactor
updating the method names in the session factory builder as a result of our changes

### CODE REVIEW

File: languages/csharp/AppEncryption/AppEncryption.Tests/AppEncryption.Tests.csproj [hunk 1]
```diff
@@ -21,4 +21,7 @@
   <ItemGroup Label="Project References">
     <ProjectReference Include="../AppEncryption/AppEncryption.csproj" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (jpaskhay) [inline languages/csharp/AppEncryption/AppEncryption.Tests/AppEncryption.Tests.csproj]: what is this change for? seems unrelated and not clear what its purpose is

File: languages/csharp/AppEncryption/AppEncryption/SessionFactory.cs [hunk 2]
```diff
@@ -45,9 +45,9 @@ public class AppEncryptionSessionFactory : IDisposable
         public interface IMetastoreStep
         {
             // Leaving this in here for now for user integration test convenience. Need to add "don't run in prod" checks somehow
-            ICryptoPolicyStep WithMemoryPersistence();
+            ICryptoPolicyStep WithInMemoryMetastore();
 
-            ICryptoPolicyStep WithMetaStorePersistence(IMetastorePersistence<JObject> persistence);
+            ICryptoPolicyStep WithMetastore(IMetastore<JObject> persistence);
```
Comment (jpaskhay) [inline languages/csharp/AppEncryption/AppEncryption/SessionFactory.cs]: minor: replace `persistence` parameter with `metastore`

File: languages/csharp/AppEncryption/AppEncryption/SessionFactory.cs [hunk 3]
```diff
@@ -158,15 +158,15 @@ internal Builder(string productId, string systemId)
...
-            public ICryptoPolicyStep WithMemoryPersistence()
+            public ICryptoPolicyStep WithInMemoryMetastore()
             {
-                metastorePersistence = new MemoryPersistenceImpl<JObject>();
+                metastore = new InMemoryMetastoreImpl<JObject>();
                 return this;
             }
 
-            public ICryptoPolicyStep WithMetaStorePersistence(IMetastorePersistence<JObject> persistence)
+            public ICryptoPolicyStep WithMetastore(IMetastore<JObject> persistence)
```
Comment (jpaskhay) [inline languages/csharp/AppEncryption/AppEncryption/SessionFactory.cs]: minor: replace `persistence` here as well, if possible

File: languages/csharp/AppEncryption/README.md [hunk 4]
```diff
@@ -152,8 +153,8 @@ and parallel encrypt calls.
 A session factory can now be built using the components we defined above.
 
 ```c#
-AppEncryptionSessionFactory appEncryptionSessionFactory = AppEncryptionSessionFactory.NewBuilder("some_product", "some_service")
-     .WithMetastorePersistence(metastorePersistence)
+SessionFactory sessionFactory = SessionFactory.NewBuilder("some_product", "some_service")
+     .WithMetastor(metastore)
```
Comment (jpaskhay) [inline languages/csharp/AppEncryption/README.md]: missing the `e` at the end of the method

File: languages/csharp/AppEncryption/README.md [hunk 5]
```diff
@@ -184,10 +185,10 @@ completely up to the calling application for storage responsibility.
 string originalPayloadString = "mysupersecretpayload";
 
 // encrypt the payload
-byte[] dataRowRecordBytes = appEncryptionBytes.Encrypt(Encoding.UTF8.GetBytes(originalPayloadString));
+byte[] dataRowRecordBytes = sessionBytes.Encrypt(Encoding.UTF8.GetBytes(originalPayloadString));
 
 // decrypt the payload
-string decryptedPayloadString = Encoding.UTF8.GetString(appEncryptionBytes.Decrypt(newDataRowRecordBytes));
+string decryptedPayloadString = Encoding.UTF8.GetString(sessionBytes.Decrypt(newDataRowRecordBytes));
```
Comment (jpaskhay) [inline languages/csharp/AppEncryption/README.md]: similar to other typos we had, should be `dataRowRecordBytes`

File: languages/csharp/AppEncryption/AppEncryption.Tests/AppEncryption.Tests.csproj (hunk 1 above)
Comment (nikoo28) [reply languages/csharp/AppEncryption/AppEncryption.Tests/AppEncryption.Tests.csproj]: Ah...i don't know how this got added. maybe some IDE magic...would remove it...
